packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasSummary.ts

```diff
@@ -0,0 +1,252 @@
+  estimatedL2SubmissionCost: constants.Zero
+}
+
+const INITIAL_GAS_SUMMARY_RESULT: UseGasSummaryResult = {
+  estimatedL1GasFees: 0,
+  estimatedL2GasFees: 0
+}
+
+export type GasEstimationStatus =
+  | 'loading'
+  | 'success'
+  | 'error'
+  | 'unavailable'
+
+export type GasEstimationResult = {
+  estimatedL1Gas: BigNumber
+  estimatedL2Gas: BigNumber
+  estimatedL2SubmissionCost: BigNumber
+}
+
+export type UseGasSummaryResult = {
+  estimatedL1GasFees: number
+  estimatedL2GasFees: number
+}
+
+export function useGasSummary(): {
+  gasSummary: UseGasSummaryResult
+  gasSummaryStatus: GasEstimationStatus
+} {
+  const {
+    app: { arbTokenBridge, arbTokenBridgeLoaded, selectedToken: token }
+  } = useAppState()
+  const [networks] = useNetworks()
+  const {
+    childChain,
+    childChainProvider,
+    parentChain,
+    parentChainProvider,
+    isDepositMode
+  } = useNetworksRelationship(networks)
+  const { address: walletAddress } = useAccount()
+  const [{ amount }] = useArbQueryParams()
+  const nativeCurrency = useNativeCurrency({ provider: childChainProvider })
+  const [gasSummaryStatus, setGasSummaryStatus] =
+    useState<GasEstimationStatus>('loading')
+  const [gasSummary, setGasSummary] = useState<UseGasSummaryResult>(
+    INITIAL_GAS_SUMMARY_RESULT
+  )
+
+  const amountBigNumber = useMemo(() => {
+    try {
+      const amountSafe = amount || '0'
+
+      if (token) {
+        return utils.parseUnits(amountSafe, token.decimals)
+      }
+
+      return utils.parseUnits(amountSafe, nativeCurrency.decimals)
+    } catch (error) {
+      return constants.Zero
+    }
+  }, [amount, token, nativeCurrency])
+
+  const parentChainGasPrice = useGasPrice({ provider: parentChainProvider })
+  const childChainGasPrice = useGasPrice({ provider: childChainProvider })
+
+  // Debounce the amount, so we run gas estimation only after the user has stopped typing for a bit
+  const amountDebounced = useDebouncedValue(amountBigNumber, 1_500)
+
+  const [result, setResult] = useState<GasEstimationResult>(
+    INITIAL_GAS_ESTIMATION_RESULT
+  )
+
+  const estimateGas = useCallback(async () => {
+    if (!walletAddress) {
+      return
+    }
+
+    const estimateGasFunctionParams = {
+      amount: amountDebounced,
+      address: walletAddress,
+      l2Provider: childChainProvider
+    }
+
+    let estimateGasResult: GasEstimationResult = INITIAL_GAS_ESTIMATION_RESULT
+
+    try {
+      setGasSummaryStatus('loading')
+
+      if (isDepositMode) {
+        estimateGasResult = token
+          ? await depositTokenEstimateGas({
+              ...estimateGasFunctionParams,
+              l1Provider: parentChainProvider,
+              erc20L1Address: token.address
+            })
+          : await depositEthEstimateGas({
+              ...estimateGasFunctionParams,
+              l1Provider: parentChainProvider
+            })
+      } else {
+        const partialEstimateGasResult = await withdrawInitTxEstimateGas({
```

Comment:
NIT: Just a bit weird (looking from a fresh user perspective) that when we try to make a withdrawal, the L2 gas shows 0.0001 but when we reverse the chains and try making a deposit, the L2 gas is 0.
I get it, it is coming straight from `getWithdrawalRequest` but should we prefix it with a ~ so that it doesn't come across as a bug?

<img width="300" alt="image" src="https://github.com/OffchainLabs/arbitrum-token-bridge/assets/7558499/4113c629-a2bc-448a-bc33-47a7343470b1">
<img width="300" alt="image" src="https://github.com/OffchainLabs/arbitrum-token-bridge/assets/7558499/a6fc1265-bc5b-4acd-bf0c-db9c538f6a46">